### PR TITLE
add documentation for gene conversion

### PR DESCRIPTION
**Pull request by fbaumdicker**
I have added a small documentation for gene conversion with a few illustrative examples.
Also: fixed a small bug as the discrete_genome feature has to be set before we set the gene conversion tract lengths in _msprimemodule.c.

### CODE REVIEW

**Comment by fbaumdicker** [comment]
Hmm, the lint test fails imo due to a part of the ancestry.py file that I did not change in this PR. What is the recommended procedure to handle this now? Should we merge anyway or should I apply the suggested changes from black?

File: docs/ancestry.rst [hunk 1]
```diff
@@ -748,10 +731,92 @@ rate along the genome can be combined in ``msprime``.
...
+    
+Continous genomes can also be used. In this case the parameters define
+the rate at which gene conversion events are initiated per unit of 
+sequence length and the mean of the exponentially distributed gene 
+conversion tract lengths. It is important to use a sequence_length
+that corresponds to the actual number of sites considered or to scale
+the gene conversion rate and tract length accordingly. Otherwise the
+distribution of the tract_lengths will not be correct. Note that if the 
+mean gene conversion tract length is set to 1 all tract lengths will be
+almost sure different. However, all gene conversion events should be
```
Comment (jeromekelleher) [inline docs/ancestry.rst:764]: I think these four sentences will just confuse people to be honest - these are the same considerations as you would have  when simulating recombination or mutation on continuous genome, and we don't really need to get into it in particular for GC. I think the example will make things clear enough, and the how we should interpret continuous coordinates dealt with in a separate section which considers GC, recombination and mutation at the same time.

So, I would delete everything from "...tract lenghts." to "The following example..."

File: docs/ancestry.rst [hunk 2]
```diff
@@ -748,10 +731,92 @@ rate along the genome can be combined in ``msprime``.
     import msprime
     from IPython.display import SVG
 
-.. todo:: Examples of small simulations using GC on its own
-    and mixed with recombination. Ideally we'd start with an
-    example with one GC event where we could explain what happened.
-    This might be tricky to finesse.
+Gene conversion events are defined by two parameters. 
+For discrete genomes these are:
```
Comment (jeromekelleher) [inline docs/ancestry.rst:735]: I would phrase this slightly differently to take the emphasis off discrete/continous genomes:

> Gene conversion events are defined by two parameters: the rate at which gene conversion events are initiated and the distribution of tract lengths. In the default case of discrete genome coordinates, tract lengths are drawn from a geometric distribution with mean ``gene_conversion_tract_length`` (which must be at least 1). Note that if we specify a tract length of 1, then all gene conversion tracts will have length exactly 1.

**Review by jeromekelleher — COMMENTED** [review]
LGTM @fbaumdicker, some minor comments.

I think the lint failure will go away once you fix the comment indentation.

File: docs/ancestry.rst (hunk 1 above)
Comment (fbaumdicker) [reply docs/ancestry.rst:764]: Done.

File: msprime/ancestry.py [hunk 3]
```diff
@@ -1057,12 +1057,17 @@ def sim_ancestry(
...
-        for ``gene_conversion_tract_length`` must also be specified.
-        See :ref:`sec_ancestry_gene_conversion` for usage examples
+        can be a single value (specifying a single rate over the entire
+        sequence). Currently an instance of :class:`RateMap` is not supported.
+        If provided, a value for ``gene_conversion_tract_length`` must also be
+        specified. See :ref:`sec_ancestry_gene_conversion` for usage examples
         for this parameter and how it interacts with other parameters.
-    :param gene_conversion_tract_length: TODO
+    :param gene_conversion_tract_length: The mean length of the gene conversion
+        tracts. For discrete genomes the tract lengths are geometrically
```
[the remark this answers is not in the document]
Comment (fbaumdicker) [reply msprime/ancestry.py:1066]: Ah, thanks.

File: msprime/cli.py [hunk 4]
```diff
@@ -55,7 +55,20 @@ def set_sigpipe_handler():
...
+    "Gene conversion at rate gamma where gamma depends on the defined "
+    "recombination rate rho=4*N0*r. If rho > 0, gc_recomb_ratio defines the ratio "
+    "g/r, where r is the probability per generation of crossing-over and g the "
+    "corresponding gene conversion probability. Gene conversions are initiated at "
+    "rate gamma=rho*gc_recomb_ratio = 4*N0*r*gc_recomb_ratio. If rho = 0 the gene "
+    "conversion rate is given by gamma=gc_recomb_ratio=4*N0*c where c is the rate "
+    "of gene conversion initiation between the ends of the simulated region of "
+    "length num_loci. If the recombination rate is not specified, standard "
+    "parameters are used, i.e. rho = 0 and num_loci = 1. The length of the gene "
+    "conversion tracts is geometric distributed with mean tract_length. "
```
[the remark this answers is not in the document]
Comment (fbaumdicker) [reply msprime/cli.py:68]: I often make this mistake. Thanks!

File: docs/ancestry.rst (hunk 2 above)
Comment (fbaumdicker) [reply docs/ancestry.rst:735]: Yes, that is better. Changed it.

**Comment by fbaumdicker** [comment]
You were right, the lint failure is now gone.

**Review by jeromekelleher — APPROVED** [review]
Great, thanks @fbaumdicker!

**Comment by jeromekelleher** [comment]
@mergifyio rebase

**Comment by jeromekelleher** [comment]
Hmm, mergify is being a bit flakey. @fbaumdicker would you mind doing a quick rebase here to bring the branch up to date please?

**Comment by fbaumdicker** [comment]
I have rebased and pushed it now. I hope this helps.

**Pull request merged**